### PR TITLE
Switch to basic authentication for registry

### PR DESCRIPTION
**Pull request by reasonerjt**
1. Add basic authorizer for registry which modify the request
to add basic authorization header to request based on configuration.
2. Set basic auth header for proxy when accessing registry
3. Switche the registry to use basic auth by default and use the basic
authorizer to access Harbor.
4. Make necessary change to test cases, particularly
"test_robot_account.py" and "docker_api.py", because the error is
changed after siwtched to basic auth from token auth.  #10604 is opened
to track the follow up work.

Signed-off-by: Daniel Jiang <jiangd@vmware.com>

### CODE REVIEW

File: src/server/middleware/v2auth/auth.go [hunk 1]
```diff
@@ -62,6 +68,9 @@ func (rc *reqChecker) check(req *http.Request) error {
 		}
 	} else if len(middleware.V2CatalogURLRe.FindStringSubmatch(req.URL.Path)) == 1 && !securityCtx.IsSysAdmin() {
 		return fmt.Errorf("unauthorized to list catalog")
+	} else if req.URL.Path == "/v2/" && !securityCtx.IsAuthenticated() {
+		ctx := context.WithValue(req.Context(), middleware.SkipInjectRegistryCredKey, true)
+		*req = *(req.WithContext(ctx))
```
Comment (ywk253100) [inline src/server/middleware/v2auth/auth.go]: We should return 401 and the supported auth method here or in our own handler.

Comment (reasonerjt) [reply src/server/middleware/v2auth/auth.go]: Good point, but IMO we may defer this, we can always make sure the registry setting is consistent with the auth setting as the communication between registry and docker or other container runtime is very well tested.

Opened #10610 but it's likely to be deferred to future.

File: src/server/middleware/v2auth/auth.go [hunk 2]
```diff
@@ -98,16 +113,24 @@ func getAction(req *http.Request) rbac.Action {
...
 
 // Middleware checks the permission of the request to access the artifact
 func Middleware() func(http.Handler) http.Handler {
+	once.Do(func() {
+		if checker.pm == nil { // for UT, where pm has been set to a mock value
+			checker = reqChecker{
+				pm: config.GlobalProjectMgr,
+			}
+		}
+	})
```
Comment (heww) [inline src/server/middleware/v2auth/auth.go]: Changing the value of the `checker` at the beginning of the UT doesn't work ? If it doesn't work, the right place of `once.Do` is  in `check` method of `reqChecker` not in the middleware.

Comment (reasonerjt) [reply src/server/middleware/v2auth/auth.go]: The problem is that config.GlobalProjectMgr is nil when checker is assigned.  So when this middleware was added to the handler I found it didn't work.

**Pull request merged**